### PR TITLE
Add jsdoc to eslint & LruMemoize.

### PR DESCRIPTION
**Pull request by aljones15**
This just adds js-doc comments and `eslint-plugin-jsdoc` to the project.

### CODE REVIEW

File: lib/LruMemoize.js [hunk 1]
```diff
@@ -3,11 +3,32 @@
...
   constructor(cacheOptions) {
     this.cache = new LRU(cacheOptions);
   }
 
+  /**
+   * Memoizes an async function in the cache.
+   *
+   * @param {object} options - Options to use.
+   * @param {string} options.key - A key for the cache.
+   * @param {Function<Promise>} options.fn - An Async Function to memoize.
```
Comment (mattcollier) [inline lib/LruMemoize.js:28]: @dlongley how do you feel about this language.  I'm thinking something more like:
`A function that returns an unresolved promise.`

Comment (dlongley) [inline lib/LruMemoize.js:28]: I think we should say this:

```suggestion
   * @param {Function<Promise>} options.fn - An Async Function that returns a promise to memoize.
```

Talking about whether or not the promise is unresolved/unsettled/etc. doesn't make sense to me. But it is the promise that is being memoized, not the function.

File: lib/LruMemoize.js [hunk 2]
```diff
@@ -3,11 +3,33 @@
...
+ *
+ * @returns {LruMemoize} The class.
+*/
 export default class LruMemoize {
   constructor(cacheOptions) {
     this.cache = new LRU(cacheOptions);
   }
 
+  /**
+   * Memoizes a promise in the cache.
```
Comment (dlongley) [inline lib/LruMemoize.js:24]:
```suggestion
   * Memoizes a promise via an LRU cache.
```

File: lib/LruMemoize.js [hunk 3]
```diff
@@ -22,6 +44,7 @@ export default class LruMemoize {
     try {
       result = await promise;
     } catch(e) {
+      // if the promise fails to resolve delete it.
```
Comment (dlongley) [inline lib/LruMemoize.js:47]:
```suggestion
      // if the promise rejects, delete it
```

**Review by dlongley — APPROVED** [review]
Approving w/some suggestions to apply. Thanks!

File: lib/LruMemoize.js (hunk 1 above)
Comment (aljones15) [reply lib/LruMemoize.js]: Went with Longley's wording here.

**Review by mattcollier — APPROVED** [review]
thanks!

**Pull request merged**